### ISSUE TITLE
Proposal on moving these large binary files to a proper file storage service

### ISSUE THREAD

**Issue by yqszxx** [plctlab/PLCT-Open-Reports#43]
AFAIK git is good at managing text based files and mainly for version control. And it seems we are using this repo as a file storage service of big binary files. When someone wants to add something to this repo, it's a must for her/him to clone or at least pull all irrelevant big chunks before addition can be commited, which makes no sense to me.

So is that better we move the contents of this repo to a proper file storage service like S3?

**Comment by lazyparser** [comment]
I've noticed too.

currently this repo is just used for saving slides. Maybe we'd find a better way for sharing. and this is not a priority IMO.

**Comment by lazyparser** [comment]
and S3 is not a good candidate, for it has been (randomly) blocked by gfw.

**Comment by lazyparser** [comment]
> and S3 is not a good candidate, for it has been (randomly) blocked by gfw.

or, we could set up two separate infrastructure for mainland China and the rest of the world, just like many giants doing now.

I have no good idea about this issue. Let's leave it open.

**Comment by lazyparser** [comment]
the another detail you may noticed is that github do provides a preview for pdf files. so the user can just open the url and watch the pdf in the browser.

sadly, the user in mainland China could not see the pdf preview without pain. the content of preview is stored on a sub domain of github, like S3.

**Comment by wsong83** [comment]
I think this is a really good suggestion and something can be done right now with some extra work.
My suggestion is to make a new release every month/week depending on the work load tolerable.
In every release, a maintainer add extra binary files (actually any types of files) as a bundle in the release (see https://github.com/riscv/riscv-isa-manual/releases)
If you pay attention, the download link of the bundled files are predictable; therefore, student can make a hyper link to the downloadable file even when the file is not bundled yet. The actual file must be sent to the maintainer by some other means, of course.
In theory, this procedure can even be automated using GitHub API?
And I think it is with the same complexity along with using Amazon S3 but with the benefit of using the same platform.

Even better, GitHub claims "Each file included in a release must be under 2 GB. There is no limit on the total size of a release, nor bandwidth usage."  https://docs.github.com/en/github/administering-a-repository/about-releases#storage-and-bandwidth-quotas

**Comment by lazyparser** [comment]
Since the repo is getting more and more big, it may be the time to follow Boyang & Wei Song's suggestion :)

I'll find a way out.

**Comment by yqszxx** [comment]
How about GitHub Pages? Use it to host the front page and use `raw` link for binaries.

**Comment by lazyparser** [comment]
经过了两年多的发展（咕），已经没有人关心/担心这个问题了。毕竟单个文件数量不到50M的话还是可以的2333

假装已经 irrelevant

感谢各位小伙伴的建议！
